cmake/packaging_test_project/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+project("Devkit-Test")
+
+cmake_minimum_required(VERSION 3.18)
+
+set(CMAKE_C_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
+
+# MUST SET THIS W/ CORRECT EXTRACTION PATH
+set(PlaidML_DIR /path/to/PlaidML-Package/share/PlaidML/cmake)
+find_package(PlaidML CONFIG REQUIRED)
+
+add_executable(edsl_test edsl_tests.cc)
+target_link_libraries(edsl_test PlaidML::plaidml_edsl_edsl)
+target_link_libraries(edsl_test PlaidML::plaidml_exec_exec)
+target_link_libraries(edsl_test PlaidML::plaidml_op_op)
+target_link_libraries(edsl_test PlaidML::plaidml_testenv)
+target_link_libraries(edsl_test PlaidML::pmlc_util_util)
+target_link_libraries(edsl_test PlaidML::gmock)
+target_link_libraries(edsl_test PlaidML::gtest)
+target_link_libraries(edsl_test PlaidML::half)
+target_link_libraries(edsl_test PlaidML::LLVMSupport)
```

Comment:
Users should link only to a single `PlaidML` dependency, and this dependency should be the `libplaidml.so` + header files.

Reply:
For gmock/gtest, we should include our own FetchContent directly, we should not include gmock/gtest as part of our package. For the other dependencies, they are all internal to `libplaidml.so`, with the exception of `plaidml_testenv`. This is the one part that is currently internal but we could refactor it to make it be external.

Reply:
Yes sorry the packaging_test_project is outdated now; it is still set up to work with the old package with all the static libraries/3rd party stuff included. Currently working on updating that to reflect the new package.

Reply:
Currently, testenv uses LLVM, which makes it hard to export to devkit users. We could refactor testenv to remove this dependency. The half dependency is likely not easy to remove, so we could also include FetchContent in the devkit cmake so that users can fetch it directly (again we should not distribute our own version of half in our package).

The only contents for our package should be:
* CMakeLists.txt
* plaidml/*.h  (for core, edsl, exec, and op) (and only the public headers)
* plaidml/edsl/edsl_test.cc
* plaidml/edsl/edsl_test.py
* libplaidml.so
* libomp.so
* libmlir_runner_utils.so.12git

In the README we should say, the following system packages are required (for ubuntu, package names may vary for other distros):
* libtinfo5
* zlib1g

The CMakeLists.txt needs to do FetchContent for:
* gflags
* googletest (include gmock and gtest)
* half

Reply:
Oops and the package needs to include the `testenv` source and header.